rust/lance-encoding/src/decoder.rs

```diff
@@ -788,6 +788,12 @@ impl CoreFieldDecoderStrategy {
             let scheduler = self.create_primitive_scheduler(field, column_info, buffers)?;
             return Ok(scheduler);
         } else if data_type.is_binary_like() {
+            if column_infos.next().is_none() {
+                return Err(Error::InvalidInput {
+                    source: "column_info cannot be none".into(),
```

Comment:
This error is very unexpected, so we should make sure to add as much context as we can.

I would recommend adding the `field` data type and name as part of the error message.